### ISSUE TITLE
Missing attributes on the Address interface

### ISSUE THREAD

**Issue by petertonysmith94** [FuelLabs/fuels-ts#1871]
### Summary

Add the follow attributes to [`AbstractAddress]`, so that types can be used as expected:
- bech32Address
- toEvmAddress
- toAssetId

### Reproduction

Example use with Wallet address
```ts
import type { Address, WalletLocked, WalletUnlocked } from 'fuels';
import { FUEL_NETWORK_URL, Provider, Wallet } from 'fuels';

const myWallet: WalletUnlocked = Wallet.generate({ provider });
const address: Address = myWallet.address;
const someWallet: WalletLocked = Wallet.fromAddress(address, provider);
```
 
Received the following error
```sh
Type 'AbstractAddress' is missing the following properties from type 'Address': bech32Address, toEvmAddress, toAssetId
ts(2739)
```

**Comment by arboleya** [comment]
In the long run, this might not make sense. 🤔 

Please check what I said [here](https://github.com/FuelLabs/fuels-ts/discussions/1844#discussioncomment-8770182).

**Comment by petertonysmith94** [comment]
That makes sense!

This issue will be resolved by the removal of `AbtractAddress` - see #1584

**Comment by petertonysmith94** [comment]
Closed in favour of #1584
